pandas/core/indexes/datetimes.py

```diff
@@ -105,8 +105,11 @@ def _dt_index_cmp(opname, cls, nat_result=False):
 
     def wrapper(self, other):
         func = getattr(super(DatetimeIndex, self), opname)
-        if (isinstance(other, datetime) or
-                isinstance(other, compat.string_types)):
+        if isinstance(other, (datetime, compat.string_types)):
+            if isinstance(other, compat.string_types):
```

Comment:
would put this block outside the top-level if, then you know that you have datetimes